### PR TITLE
[RPC] Update build support for cross compiling apps/cpp_rpc with OpenCL

### PR DESCRIPTION
**Pull request by csullivan**
Standardize build support for building and cross compiling apps/cpp_rpc with cmake.
* Add cmake coverage for building the C++ RPC server binary and update documentation.
* Add support for linking against custom OpenCL SDK employing a custom find_opencl macro. This can be useful when cross compiling with a custom OpenCL device driver.

### CODE REVIEW

File: apps/cpp_rpc/README.md [hunk 1]
```diff
@@ -19,24 +19,31 @@
...
-  if the target os is Android, you doesn't need to pass OS argument.
-  You could cross compile the TVM runtime like this:
-```
-  cd tvm
-  mkdir arm_runtime
-  cp cmake/config.cmake arm_runtime
-  cd arm_runtime
-  cmake .. -DCMAKE_CXX_COMPILER="/path/to/cross compiler g++/"
-  make runtime
+- Configure the tvm cmake build with `config.cmake` ensuring that `USE_CPP_RPC` is set to `ON` in the config.
```
Comment (FrozenGene) [inline apps/cpp_rpc/README.md:22]: I think we should tell users how to cross compile C++ rpc for embed linux platform (like Ubuntu / rasp), not only android.

**Comment by tqchen** [comment]
cc @FrozenGene

**Comment by FrozenGene** [comment]
ping @csullivan

**Comment by csullivan** [comment]
@FrozenGene Thanks for the good suggestions. I updated the documentation and build system for embedded linux to use the same flow as previously described the docs.

**Comment by FrozenGene** [comment]
Thanks @csullivan I verified it and it works.

**Pull request merged**